aggregator/data.go

```diff
@@ -513,6 +517,76 @@ func (a *Aggregator) processL7(d l7_req.L7Event) {
 	}()
 }
 
+func (a *Aggregator) processThroughputEvent(e throughput.ThroughputEvent) {
+	// filter out localhost connections
+	if e.SAddr == "127.0.0.1" || e.DAddr == "127.0.0.1" {
+		return
+	}
+
+	pkt := datastore.Packet{
+		Time:     e.Timestamp,
+		Size:     e.Size,
+		FromIP:   e.SAddr,
+		FromPort: e.SPort,
+		ToIP:     e.DAddr,
+		ToPort:   e.DPort,
+	}
+
+	// determine source information
+	a.clusterInfo.mu.RLock()
+	fromPodUid, ok := a.clusterInfo.PodIPToPodUid[e.SAddr]
+	a.clusterInfo.mu.RUnlock()
+	if ok {
+		pkt.FromType = datastore.PodSource
+		pkt.FromUID = string(fromPodUid)
+	} else {
+		pkt.FromType = datastore.OutsideSource
+		remoteDnsHost, err := getHostnameFromIP(e.SAddr)
+		if err == nil {
+			pkt.FromUID = remoteDnsHost
+		}
+	}
+
+	// determine dest information
+	a.clusterInfo.mu.RLock()
+	toPodUid, ok := a.clusterInfo.PodIPToPodUid[e.DAddr]
+	a.clusterInfo.mu.RUnlock()
+	if ok {
+		pkt.ToType = datastore.PodDest
+		pkt.ToUID = string(toPodUid)
+	} else {
+		a.clusterInfo.mu.RLock()
+		toSvcUid, ok := a.clusterInfo.ServiceIPToServiceUid[e.DAddr]
+		a.clusterInfo.mu.RUnlock()
+		if ok {
+			pkt.ToType = datastore.ServiceDest
+			pkt.ToUID = string(toSvcUid)
+		} else {
+			pkt.ToType = datastore.OutsideDest
+			remoteDnsHost, err := getHostnameFromIP(e.DAddr)
+			if err == nil {
+				pkt.ToUID = remoteDnsHost
+			}
+		}
+	}
+
+	// determine direction
+	if pkt.FromType == datastore.PodSource && pkt.ToType == datastore.PodDest || pkt.ToType == datastore.ServiceDest {
+		pkt.Direction = datastore.Internal
+	} else if pkt.FromType == datastore.PodSource && pkt.ToType == datastore.OutsideDest {
+		pkt.Direction = datastore.Egress
+	} else {
+		pkt.Direction = datastore.Ingress
+	}
+
+	go func() {
+		err := a.ds.PersistPacket(pkt)
+		if err != nil {
+			log.Logger.Error().Err(err).Msg("error persisting packet info")
+		}
+	}()
```

Comment:
I don't think there is any particular reason. Just to be fast I guess. I was mostly just copying how they process the other things as well.

Reply:
From what I can tell, the `processThroughputEvent` method was invoked in its own fire-and-forget go-routine (line 189 above), so it would seem firing `a.ds.PersistPackate` in a goroutine here doesn't buy us much.

Reply:
Good call. Just pushed a change to remove this goroutine.